### ISSUE TITLE
Mouse stuck in a small rectangle in the center of the screen

### ISSUE THREAD

**Issue by notpeelz** [ValveSoftware/gamescope#1141]
https://github.com/ValveSoftware/gamescope/assets/920910/fac211b7-e7bf-4ef8-8595-106e253a9463

I bisected from 3.14.0 (good) to 3.14.1 (bad).
First bad commit: 88eb1b477d8b1efbe6d7087dcde74052dad84049
NOTE: the video was recorded with [obs-vkcapture 1.4.9](https://github.com/nowrep/obs-vkcapture). This commit also introduced the flickering you see in the recording (not visible in-game).

Distro: Arch Linux
Compositor: swayfx 0.3.2 (based on sway 1.8.1)
Gamescope: 3.14.1 (nested)
GPU: AMD RX 6900 XT
Proton: GE-Proton8-31
Steam launch params: `gamemoderun gamescope -W 2560 -H 1440 -r 240 -- %command%`
Env vars:
```
# I've tried unsetting/reverting all of these. It doesn't seem to make a difference.
DISABLE_GAMESCOPE_WSI=1
SDL_VIDEODRIVER=x11
MANGOHUD=1
OBS_VKCAPTURE=1
```

**Comment by l12436** [comment]
I also has effect by that commit, some game mouse grab is complete fail.

**Comment by notpeelz** [comment]
Fixed by a5d853accd176d8ff8bb4c5d54b4b4cd8053b4d7, however the flickering issue remains (split off into #1159).
